src/__tests__/pinecone.test.ts

```diff
@@ -136,12 +136,31 @@ describe('Pinecone', () => {
       });
 
       test('should throw an error if required environment variable is not set', () => {
-        delete process.env.PINECONE_API_KEY;
         expect(() => new Pinecone()).toThrow(
           "Since you called 'new Pinecone()' with no configuration object, we attempted to find client configuration in environment variables but the required environment variables were not set. Missing variables: PINECONE_API_KEY. You can find the configuration values for your project in the Pinecone developer console at https://app.pinecone.io"
         );
       });
     });
+
+    test('should log a warning if the SDK is used in a browser context', () => {
+      // Mock window simulate browser context
+      const mockWindow = {} as any;
+      global.window = mockWindow;
+
+      const warnSpy = jest
+        .spyOn(console, 'warn')
+        .mockImplementation(() => null);
+
+      new Pinecone({ apiKey: 'test-api-key' });
+
+      expect(warnSpy).toHaveBeenCalledWith(
+        'Warning: The Pinecone SDK is intended for server-side use only. Using the SDK within a browser context in a production setting could lead to exposing your API key to third parties. If you have deployed the SDK to production in a browser, please rotate your API keys.'
```

Comment:
I'd change the warning language to match the updated `note` in the README: https://github.com/pinecone-io/pinecone-ts-client/pull/240/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R931